packages/app/src/cli/models/extensions/extension-instance.ts

```diff
@@ -194,18 +193,14 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
 
   async deployConfig({
     apiKey,
-    developerPlatformClient,
     appConfiguration,
   }: ExtensionDeployConfigOptions): Promise<{[key: string]: unknown} | undefined> {
-    if (this.isFunctionExtension) return this.functionDeployConfig({apiKey, developerPlatformClient, appConfiguration})
+    if (this.isFunctionExtension) return this.functionDeployConfig(apiKey)
     return this.commonDeployConfig(apiKey, appConfiguration)
   }
 
-  async functionDeployConfig({
-    apiKey,
-    developerPlatformClient,
-  }: ExtensionDeployConfigOptions): Promise<{[key: string]: unknown} | undefined> {
-    const {moduleId} = await uploadWasmBlob(this.localIdentifier, this.outputPath, developerPlatformClient)
+  async functionDeployConfig(apiKey: string): Promise<{[key: string]: unknown} | undefined> {
+    const moduleId = await randomUUID()
     return this.specification.deployConfig?.(this.configuration, this.directory, apiKey, moduleId)
   }
```

Comment:
We need to try and remove this function completely (`functionDeployConfig`) and make the moduleId generation part of the `function.ts` specification

Reply:
Good catch, yes that is straight forward enough.